calyx-backend/src/xilinx/idl.rs

```diff
@@ -0,0 +1,148 @@
+use crate::traits::Backend;
+use calyx_ir as ir;
+use calyx_utils::CalyxResult;
+use serde::Serialize;
+/// Backend that generates the Interface Design Language
+/// that we need to create AXI wrappers on arbitrary programs
+#[derive(Default)]
+pub struct IdlBackend;
+
+/// The root element of the `kernel.xml` file that describes an `.xo` package for the
+/// Xilinx toolchain, as documented [in the Vitis user guide][ug].
+///
+/// [ug]: https://docs.xilinx.com/r/en-US/ug1393-vitis-application-acceleration/RTL-Kernel-XML-File
+#[derive(Serialize)]
+struct ProgramInterface<'a> {
+    toplevel_name: &'a str,
```

Comment:
Since we're defining a little language here, I will allow myself to bikeshed the meaningless details: how about just `toplevel` instead of `toplevel_name`? Since it's a string, it's pretty clear already that it's a name.